app/core/EngineService/EngineService.ts

```diff
@@ -123,7 +123,7 @@ class EngineService {
     controllers.forEach((controller) => {
       const { name, key = undefined } = controller;
       const update_bg_state_cb = () => {
-        if (engine.context.KeyringController.vault) {
+        if (!engine.context.KeyringController.vault) {
           Logger.message('keyringController missing for UPDATE_BG_STATE_KEY');
```

Comment:
Should we change the message for 
```suggestion
          Logger.message('keyringController vault missing for UPDATE_BG_STATE_KEY');
```
?